src/rust/engine/workunit_store/src/metrics.rs

```diff
@@ -35,7 +35,12 @@ pub enum Metric {
   LocalCacheReadErrors,
   LocalCacheWriteErrors,
   LocalExecutionRequests,
+  RemoteExecutionErrors,
```

Comment:
This one is incremented for every RPC error, rather than necessarily for every corresponding `RemoteExecutionRequest`, which means it's sortof at a different level of abstraction. It might be good to namespace them differently and have these both be paired: "RPC attempt, success/fail" vs "execution attempt, success/fail"?

One thing that helps to encourage this in a lot of metrics APIs is to have a context-manager/wrapper that records both the attempt, and the success/failure as a pair. That's a little bit more challenging to use here since some of the "errors" are actually recoverable, and thus not counted that way, but just a thought.

Reply:
Good idea. I already increment a counter per RPC (`RemoteExecutionMethodExecute` and `RemoteExecutionMethodWaitExecution` depending on which method it is). I can rename those s/Method/RPC/ and have the error be "RPCError."

Then I can move the overall execution request counters (with its own error counter) up into `remote::CommandRunner`.